gourmet/check_encodings.py

```diff
@@ -27,8 +27,8 @@ def __init__ (self, file, encodings=None):
         if get_prefs().get('utf-16',False):
             self.encodings.extend(['utf_16','utf_16_le','utf_16_be'])
         if encodings: self.encodings = encodings
-        if type(file) in [str,str]:
-            file = open(file,'r')
+        if isinstance(file, str):
+            file = open(file, 'rb')
```

Comment:
I understand that there's a bytes read, because there's a bunch of decode attempts at a later stage.

This class is used, currently, only in a single place ([`gourmet/plugins/import_export/plaintext_plugin/plaintext_importer_plugin.py`](https://github.com/kirienko/gourmet/blob/d637e7a5eecd47fa21383e2f7a5ce8b25054b355/gourmet/plugins/import_export/plaintext_plugin/plaintext_importer_plugin.py#L31)), via the following path:

[`plugins/.../PlainTextImporter.do_run`](https://github.com/kirienko/gourmet/blob/d637e7a5eecd47fa21383e2f7a5ce8b25054b355/gourmet/plugins/import_export/plaintext_plugin/plaintext_importer_plugin.py#L31) -> [`check_encoding.get_file`](https://github.com/kirienko/gourmet/blob/d637e7a5eecd47fa21383e2f7a5ce8b25054b355/gourmet/check_encodings.py#L84) -> [`check_encoding.GetFile.__init__`](https://github.com/kirienko/gourmet/blob/d637e7a5eecd47fa21383e2f7a5ce8b25054b355/gourmet/check_encodings.py#L69) -> `check_encoding.CheckEncoding.__init__` (here).

As such, I propose to pass a file name in the plaintext importer, and open it only here.

I let you see whether to do that change now or later.

Reply:
I'd suggest we leave that for another PR - this is a fairly big diff already, and I was trying to stick to fairly mechanical changes which can be checked over quickly. But I'm happy to add it to this PR if you prefer.

Reply:
I'm okay with it as it is :)